src/Token/Generator.php

```diff
@@ -20,6 +20,22 @@ final class Generator implements GeneratorInterface
         Token::ALGO_HS256,
     ];
 
+    /**
+     * Minimum signing key lengths for token generation.
+     */
+    public const CONST_ALGO_MIN_LENGTHS = [
+        Token::ALGO_RS256 => 2048,
```

Comment:
(optional - we could wait until someone asks for this) Would be nice to support `RS384` - I think you could do this using `OPENSSL_ALGO_SHA384` in `openssl_sign`

(it doesn't look like you can do PS256 with `openssl_sign` - so we should ignore that for now)

Reply:
https://github.com/auth0/auth0-PHP/pull/699/commits/5ed22572c6cf185b93f72b57931b339bc0985158 Added support for RS384 to the ClientAssertionGenerator. Added support for HS384, HS512, RS384, RS512 to the generic token generator. Expanded tests to add more coverage.

I investigated `PS256` support but, as you noted, the built-in OpenSSL API doesn't support the necessary configuration to support that. I'm investigating integrating https://github.com/phpseclib/phpseclib which would allow us to support a lot more configurations in this SDK, but that will require some careful review.

Reply:
We only need to support RS256, RS384 (and PS256) currently, we don't support client_secret_jwt (HS*) - but I see no harm in leaving them there in case we do at some point